### ISSUE TITLE
nearly no TEs annotated

### ISSUE THREAD

**Issue by KamilSJaron** [clemgoub/dnaPipeTE#23]
Hello,

I run dnaPipeTE and it worked till now well. I run exactly the same read-set with exactly the same parameters

```
python3 ./dnaPipeTE.py -input "$READ_DIR"/all_reads.fastq.gz \
    -output "$OUTPUT" \
    -cpu 32 -genome_size "$GENOME_SIZE" -genome_coverage 0.5 -sample_number 3
```

and very similar config files without specified rm library 

```
rm_species = All
repeatmasker_library = 
```

However, when I look at the results, I get almost no TEs annotated by `1.3.1` in comparison to `1.2` (pasted Counts.txt files):

```
> TEs_2 # version 1.2.0
               V1        V2
1             LTR  11122701
2            LINE   1486291
3            SINE    102149
4             DNA   3251843
5            MITE         0
6        Helitron    267206
7            rRNA   6914419
8  Low_Complexity     30440
9       Satellite    512504
10 Tandem_repeats         0
11  Simple_repeat    321150
12         others     37846
13             na  87606406
14         Others         0
15          Total 129595103
> TEs_1 # version 1.3.1
               V1        V2
1             LTR         0
2            LINE     14004
3            SINE         0
4             DNA         0
5            MITE         0
6        Helitron         0
7            rRNA         0
8  Low_Complexity   1911368
9       Satellite         0
10 Tandem_repeats         0
11  Simple_repeat  19292537
12         others         0
13             na  88947189
14         Others         0
15          Total 129595357
```

The only thing I could think of is a difference in rm database. I could find this in log of run 1.3.1: `Master RepeatMasker Database: /.../RepeatMaskerLib.embl ( Complete Database: dc20170127 )`m but I don't have a log from the `1.2` run anymore. I suppose that it used the analogical database of `1.2`.

Where do you think could be a problem?

Best,
Kamil

P.S. We cited you already in our preprint, here we jsut add some data for a review :-)

**Comment by KamilSJaron** [comment]
Related question, if we have an older installation `1.2`. Is it possible to link the old rm library? Can we just specify `.../dnaPipeTE/1.2/bin/RepeatMasker/Libraries/RepeatMaskerLib.embl`in the `conf.ini` file? Is it the same format?

**Comment by clemgoub** [comment]
Hello,

Indeed the latest RM version is different. I havent try what you suggest but this may work. 
Alternatively, you can specify directly the RM library in fasta. You have to locate a file called 'specieslib' in the Library folder of RM. Something like: `/path/RepeatMasker/Libraries(y)/dc20170127/root/specieslib

use the -lib option and the path to this file in the dnaPipeTE command line.

Please keep me updated!

Cheers,

Clément

**Comment by KamilSJaron** [comment]
We tried to soft link the older RM database and it seems that it actually worked as expected. In log it says:
```
Master RepeatMasker Database: .../dnaPipeTE/1.3.1/bin/RepeatMasker/Libraries/RepeatMaskerLib.embl ( Complete Database: 20150807 )
```

However, the Counts are now completely empty:

```
LTR     0
LINE    0
SINE    0
DNA     0
MITE    0
Helitron        0
rRNA    0
Low_Complexity  0
Satellite       0
Tandem_repeats  0
Simple_repeat   0
others  0
na      86848913
Others  0
Total   129595092
```

I will try to run it with the manually linked file. Also, is there a way how to redo only the last step? Right now I recalculate everything which takes quite a bit of time...

**Comment by KamilSJaron** [comment]
Would it be heplful to upload you here the log?

**Comment by KamilSJaron** [comment]
Hello, It seems like must another problem associated with it. When I use following parameter `-RM_lib /<fullpath>/dnaPipeTE/1.2/bin/RepeatMasker/Libraries/20150807/root/specieslib`, I get literally no TEs annotated (still using exactly the same sequencing reads as before):

```
LTR     0
LINE    0
SINE    0
DNA     0
MITE    0
Helitron        0
rRNA    0
Low_Complexity  0
Satellite       0
Tandem_repeats  0
Simple_repeat   0
others  0
na      87898784
Others  0
Total   129595266
```

Well, looking more closely to the error log:

```
awk: fatal: cannot open file `.../Trinity_run0/chrysalis/readsToComponents.out.sort' for reading (No such file or directory) 
cat: .../dnaPipeTE/1.3.1/bin/RepeatMasker/Libraries/RepeatMasker.lib: No such file or directory
sort: cannot read: .../blast_out/reads_vs_annoted.blast.out: No such file or directory
rm: cannot remove ?~@~X.../blast_out/int.reads_vs_annoted.blast.out?~@~Y: No such file or directory
Error in (function (file = if (onefile) "Rplots.pdf" else "Rplot%03d.pdf",  : 
  cannot open file 'Rplots.pdf'
Calls: <Anonymous> -> print.ggplot -> grid.newpage -> <Anonymous>
Execution halted
```

(I replaced whole paths by `...` as the full addresses are just making it messier)

It seems that:

1. I did not respect the `-RM_lib=` flag somehow (still searching for the `RepeatMasker.lib`)
2. there is a pre-annotation problem too. Not sure what's that `awk` error.

**Comment by clemgoub** [comment]
I'm so sorry your message slipped through. Can you send me your whole log for this run? Did you manage to find a solution?

Best,

Clément

**Comment by KamilSJaron** [comment]
Hello, we resolved this by downgrading to `v1.2`.

I will be happy to provide you with logs to figure out where the problem was, but right now I am moving between countries. Once I will be a bit less swamped (by the end of this month or at the beginning of October) I will get back to you.

**Comment by clemgoub** [comment]
Thank you. Yes I'll be please to hear from you.
My plan is to work on an update in october, once a few dealines will be reach.

Best,

Clément
